app/models/note.rb

```
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+class Note < ApplicationRecord
+  belongs_to :agent, class_name: "User"
+  belongs_to :ticket
+
+  def self.add_note!(args)
+    agent = Agent.find_by(id: args[:agent_id])
```

What happens if `agent` is nil.

Method has a bang. So the expectation is that if something goes wrong then it will raise an exception.

However if the agent is not found then it will silently fail.

Please change `find_by` to `find_by!`.

@tejaswinichile _a

---

I was checking it manually if an agent is `present`, but yeah this makes more sense.